addons/spreadsheet_account/tests/test_account_group.py

```diff
@@ -1,80 +1,41 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+from odoo.tests import tagged
 
-from odoo.tests.common import TransactionCase
 
-
-class SpreadsheetAccountGroupTest(TransactionCase):
-    @classmethod
-    def setUpClass(cls):
-        super().setUpClass()
-
-        cls.env["account.account"].create(
-            {
-                "company_id": cls.env.user.company_id.id,
-                "name": "spreadsheet revenue Company 1",
-                "account_type": "income",
-                "code": "123",
-            }
-        )
-
-        cls.env["account.account"].create(
-            {
-                "company_id": cls.env.user.company_id.id,
-                "name": "spreadsheet expense Company 1",
-                "account_type": "income_other",
-                "code": "456",
-            }
-        )
-
-        cls.env["account.account"].create(
-            {
-                "company_id": cls.env.user.company_id.id,
-                "name": "spreadsheet revenue Company 2",
-                "account_type": "income",
-                "code": "789",
-            }
-        )
-
-    def _unlink_default_accounts(self, account_ids):
-        used_account_id = self.env["account.account"].search([("code", '=', '123')])
-        aml_ids = self.env["account.move.line"].search([("account_id", "in", account_ids.ids), ("company_id", "=", self.env.company.id)])
-        journal_ids = self.env["account.journal"].search([("default_account_id", "in", account_ids.ids), ("company_id", "=", self.env.company.id)])
-        for aml_id in aml_ids:
-            aml_id.account_id = used_account_id
-        for journal_id in journal_ids:
-            journal_id.default_account_id = used_account_id
-        for account_id in account_ids:
-            self.env["ir.property"]._set_default("property_account_income_categ_id", "product.category", used_account_id, self.env.company)
-            account_id.unlink()
+@tagged('post_install', '-at_install')
+class SpreadsheetAccountGroupTest(AccountTestInvoicingCommon):
 
     def test_fetch_account_no_group(self):
         self.assertEqual(self.env["account.account"].get_account_group([]), [])
 
     def test_fetch_account_one_group(self):
-        account_ids = self.env["account.account"].search([("account_type", "=", "income_other"), ("company_id", "=", self.env.company.id), ("code", "!=", "456")])
-        self._unlink_default_accounts(account_ids)
-        self.assertEqual(
-            self.env["account.account"].get_account_group(["income_other"]),
-            [["456"]],
-        )
+        self.assertEqual(self.env["account.account"].get_account_group(['income_other']), [['450000']])
```

Comment:
In the test with multiple accounts, you first unlink the accounts and then create two new ones.
But in this one, you use the already existing one, why the difference?

We can argue with this, but isn't it better to create a new one here too like it was done before to be 100% deterministic?